File: public/javascripts/ec/game/game_commands/capture_base.js
var CaptureBase = new JS.Class(GameCommand, {
  initialize: function(game, user, params) {
    this.callSuper();
    this.x = params.x;
    this.y = params.y;
  },

  class_name: function() {
    return 'CaptureBase';
  },

  execute: function() {
    var base = this.game.base_at(this.x, this.y);
    this.start_capture(base, this.user);
  },

  can_undo: function() {
    return true;
  }
});

File: app/game_commands/capture_base.rb
class CaptureBase < GameCommand
  def initialize(game, user, params)
    super(game, user, params)
    @x, @y = *@params.values_at(:x, :y)
  end

  def execute
    require_current_player
    require_running_game
    base = require_rival_base(@x, @y)
    unit = require_current_player_unit(@x, @y)
    raise CommandError.new("Unit cannot capture.") unless unit.can_capture?

    start_capture(base, @user)

    {}
  end
end
